data/data.go

```diff
@@ -0,0 +1,58 @@
+package data
+
+import (
+	"bytes"
+	_ "embed"
+	"encoding/csv"
+	"fmt"
+	"strconv"
+	"strings"
+)
+
+type GridIntensity struct {
+	CountryCode                  string  `json:"country_code"`
+	CountryOrRegion              string  `json:"country_or_region"`
+	Year                         int     `json:"year"`
+	LatestYear                   int     `json:"latest_year"`
+	EmissionsIntensityGCO2PerKWH float64 `json:"emissions_intensity_gco2_per_kwh"`
+}
+
+//go:embed co2-intensities-ember-2021.csv
+var data []byte
+
+func GetGridIntensityForCountry(countryCode string) (*GridIntensity, error) {
```

Comment:
@mrchrisadams I'm going to add some unit tests here but if its OK I'll do that as a follow up?

As this PR is already quite big.

Reply:
The unit test isn't as big as I thought it would be 😅 

So I've included it here but happy to split up this PR if that makes it easier to review.